lib/active_record/connection_adapters/sqlserver/database_statements.rb

```diff
@@ -281,7 +281,8 @@ def sp_executesql_sql(sql, types, params, name)
         def raw_connection_do(sql)
           case @connection_options[:mode]
           when :dblib
-            @connection.execute(sql).do
+            result = @connection.execute(sql)
+            result == false ? false : result.do
```

Comment:
I know this is essentially the same thing, but
~~~ruby
result && result.do
~~~
might be more concise.

I did not review the functionality of the change, though.  
I am just telling you an idiomatic ruby trick.

Reply:
No, it's not the same.
```
result && result.do
```
will return result if it is any falsy value.
When you look at the tiny_tds lib, you will see that it should either returns false or a result object which then can be called do on it.
In any other case - for example `nil` - I don't want it to return nil, but cause an exception, so it's easier to figure out what is going wrong.